File: javascript/7kyu/Fun with lists length.js
// https://www.codewars.com/kata/581e476d5f59408553000a4b

// Implement the method length, which accepts a linked list (head), and returns the length of the list.

function length(head) {
    var currentNode = head;
    for (var i=0;currentNode; ++i) {
        currentNode = currentNode.next
    }
    return i;
}

File: java/7kyu/Fun with lists length.java
// https://www.codewars.com/kata/581e476d5f59408553000a4b

// Implement the method length, which accepts a linked list (head), and returns the length of the list.

class Solution {
  static int length(Node head) {
    if(head == null) return 0;
    return length(head.next) + 1;
  }
}